pkg/armhelpers/groupsclient.go

```diff
@@ -18,3 +20,16 @@ func (az *AzureClient) EnsureResourceGroup(name, location string) (resourceGroup
 
 	return &response, nil
 }
+
+// DeleteResourceGroup delete the named resource group
+func (az *AzureClient) DeleteResourceGroup(name string, cancel chan struct{}) error {
+	resCh, errCh := az.groupsClient.Delete(name, cancel)
+	res := <-resCh
+	err := <-errCh
+	// When resourceGroup not exists, we will get 404
```

Comment:
should we simply return the channels to let the caller decide how to deal with them? I can argue that we may want to set a timeout instead of waiting it forever.